app/controlplane/internal/service/auth.go

```diff
@@ -260,6 +260,8 @@ type upstreamOIDCclaims struct {
 	// https://learn.microsoft.com/en-us/entra/identity/authentication/howto-authentication-use-email-signin
 	// https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference
 	PreferredUsername string `json:"preferred_username"`
+	FamilyName        string `json:"family_name"`
```

Comment:
these claims probably require the `profile` scope. Please check that it's being requested. https://learn.microsoft.com/en-us/entra/identity-platform/optional-claims-reference, https://developers.google.com/identity/openid-connect/openid-connect

Also, check that the `name` claim be just enough for this purpose (as it contains the user name in human readable format)